dashboard-ui/src/components/Survey/dynamicTemplate.jsx

```diff
@@ -0,0 +1,365 @@
+    }
+
+    toggleAccordionItem = () => {
+        this.logUserAction("toggleAccordion", `Actions Expanded`);
+    }
+
+    get question() {
+        return this.questionBase;
+    }
+    get dmName() {
+        return this.question.dmName
+    }
+    get actions() {
+        return this.question.actions
+    }
+    get supplies() {
+        return this.question.supplies
+    }
+    get situation() {
+        return this.question.situation
+    }
+    get patients() {
+        return this.question.patients
+    }
+    get decision() {
+        return this.question.decision
+    }
+    get explanation() {
+        return this.question.explanation
+    }
+
+    // Support the read-only and design modes
+    get style() {
+        return this.question.getPropertyValue("readOnly") ||
+            this.question.isDesignMode ? { pointerEvents: "none" } : undefined;
+    }
+
+    handleCloseSituationModal = () => {
+        this.setState({ showSituationModal: false });
+        setTimeout(() => {
+            window.scrollTo(0, 0);
+        }, 25);
+    };
+
+    handleShowSituationModal = () => {
+        this.setState({ showSituationModal: true });
+    }
+
+    renderElement() {
+        const cardContainerStyle = {
+            overflowX: 'auto',
+            overflowY: 'hidden',
+            whiteSpace: 'nowrap'
+        };
+
+        const patientCardStyle = {
+            display: 'inline-block',
+            minWidth: '40%',
+            maxWidth: '40%',
+            marginRight: '0.33%',
+        };
+
+        const magnifyingGlassStyle = {
+            position: 'absolute',
+            bottom: '10px',
+            left: '10px',
+            color: 'white',
+            backgroundColor: 'rgba(0, 0, 0, 0.5)',
+            borderRadius: '50%',
+            padding: '5px',
+            cursor: 'pointer',
+            transition: 'transform 0.5s ease-in-out'
+        };
+
+        const vitalsColStyle = {
+            overflowY: 'auto'     // Scrollbar for overflowing content
+        };
+
+        const rowStyle = {
+            display: 'flex',
+            alignItems: 'start',
+            overflow: 'hidden'
+        };
+
+        // button for each patient
+        const patientButtons = this.patients.map((patient, index) => (
+            <Button
+                key={index}
+                variant={this.state.visiblePatients[patient.name] ? "primary" : "secondary"}
+                onClick={() => this.togglePatientVisibility(patient.name)}
+                className="me-1"
+            >
+                {patient.name}
+                {' '}
+                {this.state.visiblePatients[patient.name] && <CheckCircleIcon />}
+            </Button>
+        ));
+
+        const patientCards = this.patients.map((patient, index) => {
+            // render card if toggled to visible
+            if (this.state.visiblePatients[patient.name]) {
+                return (
+                    <Card key={index} style={patientCardStyle}>
```

Comment:
`index` should be used as the `key` in react components only as a last resort. Do patients have a unique id attached to them? That would be a better key.

Reply:
If I remember correctly, this is a matter of performance and stability. It's not a major issue, but at least something to watch out for.

Reply:
Good catch.  Typically I will use the index for the key but if you do something like {"patient-card-" + index}

It is only an issue if multiple objects have same key and you are trying to do something with them.

Reply:
I did not know this, I included the index as a key because I was getting a sonarqube warning about not using a key. I will change all of these to use better keys (e.g patient.name)

Reply:
Think about it as giving a div an id field, they should all be unique on the webpage.

Reply:
https://react.dev/learn/rendering-lists#rules-of-keys:~:text=throughout%20its%20lifetime.-,Pitfall,-You%20might%20be

^ React documentation specifically about this